src/main/kotlin/uk/gov/justice/digital/hmpps/oneloginuserregistry/model/repository/BookerRepository.kt

```diff
@@ -9,5 +9,7 @@ interface BookerRepository : JpaRepository<Booker, Long> {
 
   fun findByOneLoginSub(authReference: String): Booker?
 
+  fun findByReference(reference: String): Booker?
+
```

Comment:
should never be null as you have reference

Reply:
Booker?

Reply:
as discussed, throwing BookerNotFoundException if reference is not valid. Also added tests.